File: Examples/Ruby/Document/accept_all_revisions.rb
require 'aspose_words_cloud'

class WorkingWithDocument

  include AsposeWordsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @words_api = WordsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def accept_all_revisions
    file_name = "SampleWordDocument.docx"
    upload_file(file_name)

    response = @words_api.accept_all_revisions(file_name)
  end

end

workingWithDocument = WorkingWithDocument.new()
puts workingWithDocument.accept_all_revisions

File: Examples/PHP/Document/AcceptAllRevisions.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Words\WordsApi;
use Aspose\Words\AsposeApp;

class Document {

	public $wordsApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->wordsApi = new WordsApi();
	}

	public function acceptAllRevisions() {       
		// Upload file to Aspose Cloud Storage
		$fileName = "SampleWordDocument.docx";
		Utils::uploadFile($fileName);

        $result = $this->wordsApi->AcceptAllRevisions($fileName, $filename = null, $storage = null, $folder = null);
        print_r($result);
	}
}

$document = new Document();
$document->acceptAllRevisions();

?>
//ExEnd:
